tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,195 @@
+    )
+    config = kw.get("config")
+    rbd_obj = initial_rbd_config(**kw)["rbd_reppool"]
+    pool = config["rep_pool_config"]["pool"]
+    image_spec = (
+        f"{config['rep_pool_config']['pool']}/{config['rep_pool_config']['image']}"
+    )
+    cache_client = ceph_cluster.get_nodes(role="client")[0]
+    admin_node = ceph_cluster.get_nodes(role="_admin")[0]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonitorWorkflows(node=cephadm)
+    pwl = PersistentWriteAheadLog(rbd_obj, cache_client, config.get("drive"))
+    config_level, entity = get_entity_level(config)
+
+    try:
+        # Configute cache client
+        pwl.configure_cache_client()
+
+        # Configure PWL
+        pwl.configure_pwl_cache(
+            config["rbd_persistent_cache_mode"],
+            config_level,
+            entity,
+            config["cache_file_size"],
+        )
+
+        # Test to verify persistent write back cache test
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        log.info("Test to kill FIO process and restarting FIO to check cache status")
+        with parallel() as p:
+            p.spawn(run_fio, **fio_ready(config, cache_client))
+            p.spawn(kill_fio, pwl.rbd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Parallel execution of cluster operations with corresponding verify cache tests
+        with parallel() as p:
+            log.info("Test to restart mon service along with cache test")
+            p.spawn(operation, rados_obj, "restart_daemon_services", daemon="mon")
+            p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+        with parallel() as p:
+            log.info("Test to restart osd service along with cache test")
+            p.spawn(restart_osd, client_node=cache_client)
+            p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+        with parallel() as p:
+            log.info("Test to restart ceph target service along with pwl cache test")
+            p.spawn(restart_ceph_target, admin_node=admin_node)
+            p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+        mon_host = ceph_cluster.get_nodes(role="mon")[0]
+        with parallel() as p:
+            log.info("Test to remove mon service along with pwl cache test")
+            cmd = f"ceph orch host label rm {mon_host.hostname} mon"
+            cache_client.exec_command(cmd=cmd, output=True)
+            p.spawn(operation, mon_obj, "remove_mon_service", host=mon_host.hostname)
+            p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+        with parallel() as p:
+            log.info(
+                "Test to add the mon service back to the cluster with pwl cache test"
+            )
+            cmd = f"ceph orch host label add {mon_host.hostname} mon"
+            cache_client.exec_command(cmd=cmd, sudo=True)
```

Comment:
Here also, if exec_command returns out == 1 or err then you should raise an exception.

Reply:
done